sway-ir/tests/ir_to_ir/bigger_inline.in_ir

```diff
@@ -0,0 +1,54 @@
+// Based on this Sway:
+//
+// script;
+//
+// fn a(b: bool) -> u64 {
+//     let x = 10;
+//     if b {
+//         x
+//     } else {
+//         1
+//     }
+// }
+//
+// fn main() -> u64 {
+//     let mut x = 0;
+//     x = a(true);
+//     x
+// }
+
+script script {
+    fn a(b: bool) -> u64 {
+        local ptr u64 x
```

Comment:
I assume this is meant to represent the equivalent of an `alloca` which makes sense since allocas in LLVM are essentially pointers. Are you planning to represent local arrays as `local ptr [8 x u64] x` (for example)? And would that imply that you need to design a `getelementptr` instruction at some point?

Reply:
Yeah, pretty much.  In general, I was trying to get this IR branch going as quickly as possible, so I've been designing it as I go.  It's obviously heavily based on LLVM but I've tried to simplify certain things and decided it's easier to make the locals explicit rather than using `alloca` instructions.  Maybe I was wrong, but we can see, it's working so far.

I already have arrays for locals, etc.  If you take a look at https://github.com/FuelLabs/sway/blob/otrho/new_ir_module/sway-core/src/optimize.rs this is where I'm converting the AST to IR and in https://github.com/FuelLabs/sway/tree/otrho/new_ir_module/sway-core/tests/sway_to_ir I have a bunch of tests demonstrating most functionality.

I've tried to keep the memory model simple, so GEP was planned but I haven't implemented it yet, mostly because everything at this stage can be done with locals on the stack and `extract_value` takes multiple indices allowing single instruction access to nested aggregates.  But, yeah, I've been designing this as I go, and since it's internal to Sway (i.e., not used elsewhere at all) and not customer facing we can change whatever we like as needed if the design could be improved.

Reply:
Oh I see the array example now. I think that makes sense, thanks. A lot can be done with extract_value/insert_value. 
The power of the `gep` is that it lets us index the pointer (first index of the `gep) as well index the type that the pointer points to (the remaining indices), but I'm not sure we're going to need to index the pointer itself for `Sway`.